### PR TITLE
adding CancelAllSpotOrders instruction into the lib

### PR DESCRIPTION
**Pull request by godmodegalactus**
Adding initial part for setting instruction.
Will add the tests once my pull request on mango-v3 is merged so that i can test it on devnet.

### CODE REVIEW

File: test/cancelallspotorders.test.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      ),
+    )
+  );
+  const connection = new Connection(
+    config.cluster_urls[cluster],
+    'processed' as Commitment,
+  );
+  // init
+  const testGroup = new TestGroup(connection, payer, mangoProgramId);
+  const mangoGroupKey = await testGroup.init();
```
Comment (mschneider) [inline test/cancelallspotorders.test.ts:33]: why are you not using a regular mango group? e.g. devnet.2

File: test/cancelallspotorders.test.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  const market = testGroup.spotMarkets[3];
+  const oracle = testGroup.oraclePks[3];
+  
+  await mangoAccount.reload(connection);
+  const oracleData = await connection.getAccountInfo(oracle);
+  if(oracleData == null)
+    return;
+  // placing orders
+  const price = pyth.parsePriceData(oracleData?.data);
+  const assetPrice = price?.price;
```
Comment (mschneider) [inline test/cancelallspotorders.test.ts:94]: makes more sense here to use price!.price, else you'll end up posting 0 price orders

File: test/cancelallspotorders.test.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  console.log('asset price is ' + assetPrice)
+  console.log('place order 1')
+  await testGroup.client.placeSpotOrder(mangoGroup, mangoAccount, mangoGroup.mangoCache, market, payer, "buy", assetPrice ?? 0 * 0.99, 1, 'limit');
+  await mangoAccount.reload(connection);
+  console.log('place order 2')
+  await testGroup.client.placeSpotOrder(mangoGroup, mangoAccount, mangoGroup.mangoCache, market, payer, "buy", assetPrice ?? 0 * 0.98, 1, 'limit');
```
Comment (mschneider) [inline test/cancelallspotorders.test.ts:100]: is it possible to place bid and ask? it's a more relevant use. i usually place .9 bid and 1.1 ask to test quoting

**Comment by godmodegalactus** [comment]
@mschneider PR updated after the review.

File: package.json [hunk 4]
```diff
@@ -75,6 +76,7 @@
     "@project-serum/anchor": "^0.21.0",
     "@project-serum/serum": "0.13.55",
     "@project-serum/sol-wallet-adapter": "^0.2.0",
+    "@pythnetwork/client": "^2.7.0",
```
Comment (mschneider) [inline package.json:79]: should be a devDependency as it's only used in tests. We don't want to increase the bundle size of every UI using mango-client

**Pull request merged**